### PR TITLE
feat(web): add a generator to add @nrwl/web:file-server target

### PR DESCRIPTION
**Pull request by barbados-clemens**
## Current Behavior

no automated way to setup @nrwl/web:file-server to serve my static files contents 

## Expected Behavior

generator to setup a 'serve-static' (configurable) target for my project

## Related Issue(s)

this is the start of making feature based testing easier/more robust with nx.

TODO:
 - [x] e2e tests

### CODE REVIEW

File: packages/web/generators.json [hunk 1]
```diff
@@ -15,6 +15,11 @@
       "aliases": ["app"],
       "x-type": "application",
       "description": "Create an web application."
+    },
+    "static-config": {
```
Comment (barbados-clemens) [inline packages/web/generators.json:19]: Unsure what we want to name this, but figured I'd start here

File: packages/web/src/generators/static-serve/static-serve-configuration.ts [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+      if (noBracketParts.length === 2 && noBracketParts?.[1]) {
+        const key = noBracketParts[1].trim();
+        const value = buildTargetConfig.options?.[key];
+        if (value) {
+          maybeOutputValue = value;
+          break;
+        }
+      }
+    }
```
Comment (barbados-clemens) [inline packages/web/src/generators/static-serve/static-serve-configuration.ts:74]: unsure if we want to do this, but figured trying to find the output if the target doesn't have 'outputPath' would be helpful for those setting up the static target

File: docs/generated/packages/web/generators/static-config.json [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    "description": "Add a new serve target to serve a build apps static files. This allows for faster serving of the static build files by reusing the case. Helpful when reserving the app over and over again like in e2e tests.",
+    "type": "object",
+    "properties": {
+      "buildTarget": {
+        "type": "string",
+        "description": "Name of the build target to serve"
+      },
+      "outputPath": {
+        "type": "string",
+        "description": "Path to the directory of the built files. This is only optional if buildTarget doesn't specify an outputPath executor option."
```
Comment (isaacplmann) [inline docs/generated/packages/web/generators/static-config.json:18]:
```suggestion
        "description": "Path to the directory of the built files. This is only needed if buildTarget doesn't specify an outputPath executor option."
```

File: packages/web/src/generators/static-serve/schema.json [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  "description": "Add a new serve target to serve a build apps static files. This allows for faster serving of the static build files by reusing the case. Helpful when reserving the app over and over again like in e2e tests.",
+  "type": "object",
+  "properties": {
+    "buildTarget": {
+      "type": "string",
+      "description": "Name of the build target to serve"
+    },
+    "outputPath": {
+      "type": "string",
+      "description": "Path to the directory of the built files. This is only optional if buildTarget doesn't specify an outputPath executor option."
```
Comment (isaacplmann) [inline packages/web/src/generators/static-serve/schema.json:15]:
```suggestion
      "description": "Path to the directory of the built files. This is only needed if buildTarget doesn't specify an outputPath executor option."
```

**Review by isaacplmann — CHANGES_REQUESTED** [review]

**Review by mandarini — APPROVED** [review]

**Review by isaacplmann — APPROVED** [review]

**Pull request merged**